tests/blob-api.js

```diff
@@ -0,0 +1,122 @@
+import { join, basename } from 'node:path'
+import { fileURLToPath } from 'url'
+import test from 'brittle'
+import { BlobApi } from '../src/blob-api.js'
+import { createBlobServer, getPort } from '../src/blob-server/index.js'
+import { createBlobStore } from './helpers/blob-store.js'
+import { timeoutException } from './helpers/index.js'
+
+test('get port after listening event with explicit port', async (t) => {
+  const blobStore = createBlobStore()
+  const server = await createBlobServer({ blobStore })
+
+  t.ok(await timeoutException(getPort(server.server)))
+
+  await new Promise((resolve) => {
+    server.listen({ port: 3456 }, (err, address) => {
+      resolve(address)
+    })
+  })
+
+  const port = await getPort(server.server)
+
+  t.is(typeof port, 'number')
+  t.is(port, 3456)
+
+  t.teardown(async () => {
+    await server.close()
+  })
+})
+
+test('get port after listening event with unset port', async (t) => {
```

Comment:
I think this is covered in the getUrl test below without needing to unit test getPort()?